pkg/operator/controller/garden/reconciler_delete.go

```diff
@@ -174,6 +191,13 @@ func (r *Reconciler) delete(
 			Fn:           component.OpWait(gardenerResourceManager).Destroy,
 			Dependencies: flow.NewTaskIDs(ensureNoManagedResourcesExistAnymore),
 		})
+		cleanupGenericTokenKubeconfig = g.Add(flow.Task{
+			Name: "Cleaning up generic token kubeconfig",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return r.cleanupGenericTokenKubeconfig(ctx, secretsManager)
+			}),
+			Dependencies: flow.NewTaskIDs(destroyGardenerResourceManager),
+		})
```

Comment:
Why do we need to wait for this task? Can't we destroy it at the same time when the virtual garden GRM is destroyed?